static/js/api.ts

```diff
@@ -79,11 +79,24 @@ class API {
       if (!res.ok) {
         throw new Error(`${res.status}`);
       }
-
       return res.json();
     });
   }
 
+  putJsonWithEmptyResponse(url: string, data: Record<string, unknown>): Promise<void> {
+    const opts = {
+      method: 'PUT',
+      headers: this.headers('application/json'),
```

Comment:
Minor note, in my understanding `this.headers` functions also set `Accept: 'application/json'` which is not needed in this case.